cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIMongoBaseSinkTest.java

```diff
@@ -168,37 +282,89 @@ public void testBuildCollectionNameDMByServicePathRootServicePath() {
+                        + "'sth_/_someId_someType'");
+                throw e;
+            } // try catch
+        } catch (Exception e) {
+            System.out.println(getTestTraceHead("[NGSIMongoBaseSink.buildCollectionName]")
+                        + "- FAIL - There was some problem when building the collection name");
+            assertTrue(false);
+        } // catch
+    } // testBuildCollectionNameDMByEntityRootServicePathOldEncoding
     
     /**
      * [NGSIMongoBaseSink.buildCollectionName] -------- When / service-path is notified/defaulted and
-     * data_model=dm-by-entity, the MongoDB collections name is <prefix>/_<entityId>_<entityType>.
+     * data_model=dm-by-entity, the MongoDB collections name is the concatenation of the \<prefix\>, \<service-path\>,
+     * \<entityId\> and \<entityType\>.
      */
     @Test
-    public void testBuildCollectionNameDMByEntityRootServicePath() {
+    public void testBuildCollectionNameDMByEntityRootServicePathNewEncoding() {
         System.out.println(getTestTraceHead("[NGSIMongoBaseSink.buildCollectionName]")
                 + "-------- When / service-path is notified/defaulted and data_model=dm-by-entity, the MongoDB"
-                + "collections name is <prefix>/_<entityId>_<entityType>");
+                + "collections name is the concatenation of the <prefix>, <service-path>, <entityId> and <entityType>");
         String collectionPrefix = "sth_";
         String dbPrefix = "sth_";
         String dataModel = "dm-by-entity";
+        String enableEncoding = "true";
         NGSIMongoBaseSinkImpl sink = new NGSIMongoBaseSinkImpl();
-        sink.configure(CommonUtilsForTests.createContextForMongoSTH(collectionPrefix, dbPrefix, dataModel));
+        sink.configure(CommonUtilsForTests.createContextForMongoSTH(collectionPrefix, dbPrefix, dataModel,
+                enableEncoding));
         String dbName = "sth_default";
         String fiwareService = "default";
         String fiwareServicePath = "/";
-        String entity = "someId_someType";
-        String attribute = "someName_someType";
+        String entity = "someId=someType";
+        String attribute = "someName";
         boolean isAggregated = false;
         String entityId = "someId";
         String entityType = "someType";
         
         try {
             String collectionName = sink.buildCollectionName(dbName, fiwareServicePath, entity, attribute,
                     isAggregated, entityId, entityType, fiwareService);
+            String expectedCollectionName = "sth_x002fxffffsomeIdxffffsomeType";
             
             try {
-                assertEquals("sth_/_someId_someType", collectionName);
+                assertEquals(expectedCollectionName, collectionName);
                 System.out.println(getTestTraceHead("[NGSIMongoBaseSink.buildCollectionName]")
                         + "-  OK  - '" + collectionName + "' was crated as collection name");
```

Comment:
Fixed in https://github.com/telefonicaid/fiware-cygnus/pull/1133/commits/3ba31a420f9ddda138c790c11f4d8fdc426ba8f4